src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java

```diff
@@ -69,6 +72,74 @@ public UnmanagedBaseActor(
                         errorCheckFunction));
     }
 
+    public UnmanagedBaseActor(
+            String name,
+            ActorConfig config,
+            ConnectionRegistry connectionRegistry,
+            ObjectMapper mapper,
+            RetryStrategyFactory retryStrategyFactory,
+            ExceptionHandlingFactory exceptionHandlingFactory,
+            Class<? extends Message> clazz,
+            MessageHandlingFunction<Message, Boolean> handlerFunction,
+            Function<Throwable, Boolean> errorCheckFunction) {
+        val consumerConnection = connectionRegistry.createOrGet(consumerConfig(name, config.getConsumerConfig()));
+        val producerConnection = connectionRegistry.createOrGet(producerConfig(name, config.getProducerConfig()));
+        this.publishActor = new UnmanagedPublisher<>(name, config, producerConnection, mapper);
+        this.consumeActor = new UnmanagedConsumer<>(
+                name, config, consumerConnection, mapper, retryStrategyFactory, exceptionHandlingFactory, clazz, handlerFunction,
+                errorCheckFunction);
+    }
+
+    private ConnectionConfig producerConfig(String actorName, ProducerConfig producerConfig) {
+        if (producerConfig == null) {
+            return ConnectionConfig.builder()
+                    .name(Constants.DEFAULT_CONNECTION_NAME)
+                    .build();
+        }
+
+        return producerConfig.getConnectionIsolationStrategy().accept(new ConnectionIsolationStrategyVisitor<ConnectionConfig>() {
```

Comment:
Seems like same code is being repeated twice in producer and consumer? Can be refactored with isolation strategy as a param?

Reply:
Done. Two methods are still required to have different naming format but core logic has been moved to single method